testcase/testcase.go

```diff
@@ -251,12 +256,15 @@ func (tcs *TestCaseSet) Compare(events []logstash.Event, quiet bool, diffCommand
 	}()
 
 	for i, actualEvent := range events {
-		if !quiet {
-			var description string
-			if len(tcs.descriptions[i]) > 0 {
-				description = fmt.Sprintf(" (%s)", tcs.descriptions[i])
-			}
-			fmt.Printf("Comparing message %d of %d from %s%s...\n", i+1, len(events), filepath.Base(tcs.File), description)
+		comparisonResult := &logstash_observer.ComparisonResult{
+			Path:       filepath.Base(tcs.File),
+			EventIndex: i,
+			Status:     true,
+		}
+		if len(tcs.descriptions) > i {
```

Comment:
You have not guarantee that all of you test have description. So we just test that `len(tcs.descriptions[i]) > 0` and then use `tcs.descriptions[i]` you can to be in case where you are out of array (I have already to be in this scenario).

Reply:
If I put `len(tcs.descriptions[i]) > 0`, the unit test failed on `TestCompare` (testcase_test.go) because of TestCaseSet struct has not description. So we get `panic: runtime error: index out of range`

Reply:
Maybee I need to put `len(tcs.descriptions) > i && len(tcs.descriptions[i]) > 0`, aka array not empty and string not empty

Reply:
The reason the test fails is because the test is buggy and doesn't initialize the TestCaseSet struct like New does. I can clean that up later.

Reply:
Ok, so I continue to test if array is not empty and you can clean it after you fix test...